src/pages/studyView/table/FusionGenesTable.tsx

```diff
@@ -0,0 +1,310 @@
+import * as React from "react";
+import {GeneIdentifier, FusionGenesData} from "pages/studyView/StudyViewPageStore";
+import {observer} from "mobx-react";
+import styles from "./tables.module.scss";
+import {MutationCountByGene} from "shared/api/generated/CBioPortalAPIInternal";
+import LabeledCheckbox from "../../../shared/components/labeledCheckbox/LabeledCheckbox";
+import MobxPromise from "mobxpromise";
+import {If} from 'react-if';
+import * as _ from 'lodash';
+import classnames from 'classnames';
+import DefaultTooltip from "shared/components/defaultTooltip/DefaultTooltip";
+import FixedHeaderTable from "./FixedHeaderTable";
+import {action, computed, IReactionDisposer, observable, reaction} from "mobx";
+import autobind from 'autobind-decorator';
+import {
+    correctMargin,
+    correctColumnWidth,
+    getFixedHeaderNumberCellMargin,
+    getFixedHeaderTableMaxLengthStringPixel,
+    getFrequencyStr,
+    getQValue
+} from "../StudyViewUtils";
+import {SortDirection} from "../../../shared/components/lazyMobXTable/LazyMobXTable";
+import {DEFAULT_SORTING_COLUMN} from "../StudyViewConfig";
+
+export interface IFusionGenesTablePros {
+    promise: MobxPromise<FusionGenesData>;
+    width: number;
+    height: number;
+    filters: number[];
+    onUserSelection: (value: GeneIdentifier[]) => void;
+    numOfSelectedSamples: number;
+    onGeneSelect: (hugoGeneSymbol: string) => void;
+    selectedGenes: string[];
+}
+
+type FusionGenesTableUserSelectionWithIndex = {
+    entrezGeneId: number;
+    hugoGeneSymbol: string;
+    rowIndex: number;
+}
+
+enum ColumnKey {
+    GENE = 'Gene',
+    NUMBER_MUTATIONS = '# Mut',
+    NUMBER = '#'
+}
+
+class FusionGenesTableComponent extends FixedHeaderTable<MutationCountByGene> {
+}
+
+@observer
```

Comment:
@khzhu @zhx828 Just for discusion but do you think there's opportunity for reusing more code (super class?) between these different table types?  i would say we should do it only if it's obvious that we're copying and pasting a lot.  Or if there are going to be a lot more table types.

Reply:
@JJ, @alisman, @zhx828, 
how will fusion table be displayed once structure variant PR merged in, will still be as same as mutation table? 
I am re-thinking how to address Aaron's above comment to factor gene tables. CNAGenesTable and MutationGenesTable are a bit different since we have cytobands and alterations in the CNAGenesTable, while have Frequencies in the MutationGenesTable, makes sense to give them their own tables. But new fusionGenesTable is merely a copy of Mutation table, since it does not have its own data model at the moment. If FusionGenesTable is always as same as MutationGenesTable, it will make more sense to have a super class for them. 
![Screen Shot 2019-08-19 at 5 42 39 PM](https://user-images.githubusercontent.com/17375786/63301643-13dca680-c2a9-11e9-8ae4-015cad1b630a.png)

Reply:
@khzhu we may keep it the same way after using the SV table for the first iteration. In the long run, we may want to to find a way to display the partner genes, but let's not worry about it for now.

Reply:
thanks, @JJ. 
I vote for having FusionGenesTable by its own for the future so we could easily add partner genes and possible remove Freq coloumn(do not think is useful). What do you think @alisman?

Reply:
here are three tables side by side
![Screen Shot 2019-08-20 at 11 54 06 AM](https://user-images.githubusercontent.com/17375786/63363332-679ace80-c341-11e9-98a4-5fd6de82c20d.png)

Reply:
@alisman these tables are using the same component FixedHeaderTable. There is no much of space to create a super class for these study view tables since they are highly customized. But I do think there are more space to move some of the functions to utils so they can be shared.

Reply:
moved some common interface/type to TabUtil file for share.

Reply:
@alisman I merged three tables to one component. It's not a super class, since we already using FixedHeaderTable. Essentially, it's just a util component including all the column definitions.